### DISCUSSION TITLE
Error 400 when using external nginx proxy with http3

### DISCUSSION THREAD

**Discussion by bertmelis** [immich-app/immich#4345] [Show and tell]
If you, like me, have externalized the nginx proxy container and you want to use http3, you should **comment out** this line in your config:

```
proxy_set_header X-Forwarded-Proto $forwarded_protocol;
```

This to avoid http 400 errors.

I haven't seen any obvious effects on the inner workings of Immich but I haven't checked thouroughly so use with care.

**Comment by nixigaj** [comment]
Personally, I had no problem with having the `X-Forwarded-Proto` header active but instead I had to remove the `proxy_set_header Host $http_host;` header to get it to work with HTTP/3. If it is to any help for debugging, this is over a private WireGuard network with the subnet `fddd::/64`

**Reply by WatchKitty** [reply]
I encountered the same issue and arrived here by searching the keyword "IMMICH HTTP3 NGINX HTTP400" on Google. I used the Nginx example config from the official documentation at [Reverse Proxy | Immich](https://immich.app/docs/administration/reverse-proxy). In fact, the `Host` header does not need to be commented out. Changing it to `proxy_set_header Host $host;` can resolve the HTTP400 error.
The reason is: If the `Host` header is not present in the request headers, the `$http_host` variable will be empty, whereas the `$host` variable defaults to the primary domain name. Therefore, generally speaking, using `$host` instead of the `$http_host` variable can prevent errors caused by missing Host headers in HTTP requests, avoiding situations where the Host header is not properly rewritten.

**Comment by Webbeh** [comment]
Same issue here with latest version (134 iirc).
`proxy_set_header Host $host` instead of `proxy_set_header Host $http_host` did the trick for me as well.
